src/test/java/teammates/test/cases/webapi/UpdateInstructorActionTest.java

```diff
@@ -0,0 +1,201 @@
+        InstructorAttributes instructorToEdit = typicalBundle.instructors.get("instructorNotDisplayedToStudent1");
+        String instructorId = instructorToEdit.googleId;
+        String courseId = instructorToEdit.courseId;
+
+        loginAsInstructor(instructorId);
+
+        ______TS("Typical case: edit instructor successfully");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, courseId,
+        };
+
+        String newInstructorName = "newName";
+        String newInstructorEmail = "newEmail@email.com";
+
+        InstructorCreateRequest reqBody = new InstructorCreateRequest(instructorId, newInstructorName,
+                newInstructorEmail, instructorToEdit.role,
+                instructorToEdit.displayedName, instructorToEdit.isDisplayedToStudents);
+
+        UpdateInstructorAction updateInstructorAction = getAction(reqBody, submissionParams);
+        JsonResult actionOutput = getJsonResult(updateInstructorAction);
+
+        MessageOutput test = (MessageOutput) actionOutput.getOutput();
+        System.out.println(test.getMessage());
+
+        assertEquals(HttpStatus.SC_OK, actionOutput.getStatusCode());
+
+        MessageOutput msg = (MessageOutput) actionOutput.getOutput();
+        assertEquals("The changes to the instructor " + newInstructorName + " has been updated.",
+                msg.getMessage());
+
+        InstructorAttributes editedInstructor = instructorsLogic.getInstructorForGoogleId(courseId, instructorId);
+        assertEquals(newInstructorName, editedInstructor.name);
+        assertEquals(newInstructorEmail, editedInstructor.email);
+        assertTrue(editedInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
+        assertTrue(editedInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR));
+        assertTrue(editedInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION));
+        assertTrue(editedInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT));
+
+        ______TS("Failure case: edit failed due to invalid parameters");
+
+        String invalidEmail = "wrongEmail.com";
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, courseId,
+        };
+        reqBody = new InstructorCreateRequest(instructorId, instructorToEdit.name,
+                invalidEmail, Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER,
+                Const.ParamsNames.INSTRUCTOR_DISPLAY_NAME, true);
+
+        updateInstructorAction = getAction(reqBody, submissionParams);
+        actionOutput = getJsonResult(updateInstructorAction);
+
+        assertEquals(HttpStatus.SC_BAD_REQUEST, actionOutput.getStatusCode());
+
+        msg = (MessageOutput) actionOutput.getOutput();
+        String expectedErrorMessage = new FieldValidator().getInvalidityInfoForEmail(invalidEmail);
```

Comment:
The `FieldValidator` doesn't hold mutable state, so you can extract out a `static final FieldValidator` in this test class and reuse it.

Reply:
Do you mean have a class variable in this test case? 

`static final FieldValidator = new FieldValidator() //at the top of test class`

Reply:
Yes, you can do that. Make it `private` too.